cog/core/downtime.py

```
@@ -0,0 +1,97 @@
+        if self.end.tzinfo is None:
+            self.end.astimezone()
+
+    def __contains__(self, timestamp: datetime):
+        if self.end.tzinfo is None:  # XXX: True when self.end is set after init
+            self.end = self.end.astimezone()
+
+        return self.start <= timestamp <= self.end
+
+    @staticmethod
+    def from_str(
+        start_str: str, end_str: str | None = None, is_restored=False
```

Ruff (0.8.2)

Boolean default positional argument in function definition

(FBT002)